src/ripple/rpc/handlers/AccountObjects.cpp

```diff
@@ -0,0 +1,190 @@
+        Json::Value const jv(RPC::accountFromString(ledger, rippleAddress, bIndex,
+            strIdent, iIndex, false, context.netOps));
+        if (! jv.empty ())
+        {
+            for (Json::Value::const_iterator it (jv.begin ()); it != jv.end (); ++it)
+                result[it.memberName ()] = it.key ();
+
+            return result;
+        }
+    }
+
+    if (! ledger->hasAccount (rippleAddress))
+        return rpcError (rpcACT_NOT_FOUND);
+
+    unsigned int limit;
+    if (params.isMember (jss::limit))
+    {
+        auto const& jvLimit (params[jss::limit]);
+        if (! jvLimit.isIntegral ())
+            return RPC::expected_field_error (jss::limit, "unsigned integer");
+
+        limit = jvLimit.isUInt () ? jvLimit.asUInt () :
+            std::max (0, jvLimit.asInt ());
+
+        if (context.role != Role::ADMIN)
+        {
+            limit = std::max (RPC::Tuning::minObjectsPerRequest,
+                std::min (limit, RPC::Tuning::maxObjectsPerRequest));
+        }
+    }
+    else
+    {
+        limit = RPC::Tuning::defaultObjectsPerRequest;
+    }
+    
+    Account const& raAccount (rippleAddress.getAccountID ());
+    unsigned int reserve (limit);
+    uint256 startAfter;
+    std::uint64_t startHint (0);
+
+    if (params.isMember (jss::marker))
+    {
+        // We have a start point. Use limit - 1 from the result and use the
+        // very last one for the resume.
+        Json::Value const& marker (params[jss::marker]);
+
+        if (! marker.isString ())
+            return RPC::expected_field_error (jss::marker, "string");
+
+        startAfter.SetHex (marker.asString ());
+        SLE::pointer sleObj (ledger->getSLEi (startAfter));
+
+        if (sleObj == nullptr)
+            return rpcError (rpcINVALID_PARAMS);
+
+        switch (sleObj->getType ())
+        {
+        case ltRIPPLE_STATE:
+            if (sleObj->getFieldAmount (sfLowLimit).getIssuer () == raAccount)
+                startHint = sleObj->getFieldU64 (sfLowNode);
+            else if (sleObj->getFieldAmount (sfHighLimit).getIssuer () == raAccount)
+                startHint = sleObj->getFieldU64 (sfHighNode);
+            else
+                return rpcError (rpcINVALID_PARAMS);
+            
+            break;
+
+        case ltOFFER:
+            startHint = sleObj->getFieldU64 (sfOwnerNode);
+            break;
+        }
+        
+        // Caller provided the first object (startAfter), add it as first result
+        result[jss::account_objects].append (sleObj->getJson (0));
+    }
+    else
+    {
+        // We have no start point, limit should be one higher than requested.
+        ++reserve;
+    }
+
+    Json::Value jv (Json::nullValue);
+        
+    if (! context.netOps.getAccountObjects(ledger, raAccount, startAfter,
+        startHint, reserve, [&](SLE::ref sleCur)
+        {
+            if (! jv.isNull ())
+                result[jss::account_objects].append (jv);
```

Comment:
Should you null out `jv` here after you use it. I don't know if this is likely or even possible, but if the lambda returns true, setting `jv`, then false twice in a row, `jv` will be set both times, and be appended both times. Right?